specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+{
+    //! [ClientCertificateRequested2]
+    // Register a handler for the `ClientCertificateRequested` event.
+    // This example hides the default client certificate dialog and shows a custom dialog instead.
+    // The dialog box displays mutually trusted certificates list and allows the user to select a certificate.
+    // Selecting `OK` will continue the request with a certificate.
+    // Selecting `CANCEL` will continue the request without a certificate.
+    CHECK_FAILURE(m_webView->add_ClientCertificateRequested(
+            Callback<ICoreWebView2ClientCertificateRequestedEventHandler>(
+                [this](
+                    ICoreWebView2* sender,
+                    ICoreWebView2ClientCertificateRequestedEventArgs* args) {
+                    auto showDialog = [this, args] {
+                        ICoreWebView2ClientCertificateList* certificateList;
+                        CHECK_FAILURE(args->get_MutuallyTrustedCertificates(&certificateList));
+
+                        wil::unique_cotaskmem_string host;
+                        CHECK_FAILURE(args->get_Host(&host));
+
+                        INT port = FALSE;
+                        CHECK_FAILURE(args->get_Port(&port));
+
+                        UINT certificateListCount = 0;
+                        CHECK_FAILURE(certificateList->get_Count(&certificateListCount));
+
+                        ICoreWebView2ClientCertificate* certificate = nullptr;
+
+                        ClientCertificate client_certificate;
+                        for (UINT i = 0; i < certificateListCount; i++)
+                        {
+                            CHECK_FAILURE(certificateList->GetValueAtIndex(i, &certificate));
+
+                            wil::unique_cotaskmem_string subject;
+                            CHECK_FAILURE(certificate->get_Subject(&subject));
+                            client_certificate.Subject = subject.get();
+
+                            wil::unique_cotaskmem_string displayName;
+                            CHECK_FAILURE(certificate->get_DisplayName(&displayName));
+                            client_certificate.DisplayName = displayName.get();
+
+                            wil::unique_cotaskmem_string issuer;
+                            CHECK_FAILURE(certificate->get_Issuer(&issuer));
+                            client_certificate.Issuer = issuer.get();
+
+                            wil::unique_cotaskmem_string serialNumber;
+                            CHECK_FAILURE(certificate->get_SerialNumber(&serialNumber));
+
+                            COREWEBVIEW2_CLIENT_CERTIFICATE_KIND certificateKind;
+                            CHECK_FAILURE(certificate->get_CertificateKind(&certificateKind));
+                            client_certificate.CertificateKind = NameOfCertificateKind(certificateKind);
+
+                            double startDate;
+                            CHECK_FAILURE(certificate->get_ValidStartDate(&startDate));
```

Comment:
[Certificate](https://docs.microsoft.com/uwp/api/Windows.Security.Cryptography.Certificates.Certificate) calls this `ValidFrom` (and corresponding `ValidTo`').